### PR TITLE
feat: v8 Bot

### PR DESCRIPTION
**Pull request by mehah**
Sponsored By:
@luanluciano93
@SkullzOTS
@kokekanon
@FranciskoKing
@Kizuno18

### CODE REVIEW

File: modules/game_bot/default_configs/vBot_4.7/vBot/extras.lua [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+        voc = "ED"
+      elseif text:lower():find("knight") then
+        voc = "EK"
+      elseif text:lower():find("paladin") then
+        voc = "RP"
+      end
+      local creature = getCreatureByName(name)
+      if creature then
+        creature:setText("\n" .. level .. voc .. "\n" .. guild)
+      end
```
Comment (kokekanon) [inline modules/game_bot/default_configs/vBot_4.7/vBot/extras.lua]:
```suggestion
      local voc
      if text:lower():find("sorcerer") then
        voc = "MS"
      elseif text:lower():find("druid") then
        voc = "ED"
      elseif text:lower():find("knight") then
        voc = "EK"
      elseif text:lower():find("paladin") then
        voc = "RP"
      end
      if not voc then
        return false
      end
      local creature = getCreatureByName(name)
      if creature then
        creature:setText("\n" .. level .. voc .. "\n" .. guild)
      end
```

Comment (kokekanon) [reply modules/game_bot/default_configs/vBot_4.7/vBot/extras.lua]: if not voc then
        return false
      end

fix this in custom server or old protocol

`ERROR: [BOT] [string "/vBot/extras.lua"]:608: attempt to concatenate local 'voc' (a nil value)`

File: modules/game_interface/interface.otmod [hunk 2]
```diff
@@ -36,5 +37,7 @@ Module
     - game_unjustifiedpoints
     - game_shaders
     - game_attachedeffects
+    - client_profiles
+    - game_bot
```
Comment (kokekanon) [inline modules/game_interface/interface.otmod]:
```suggestion
    - game_bot
    - game_itemselector
```

Comment (kokekanon) [reply modules/game_interface/interface.otmod]: fix click on this
image

File: modules/game_bot/default_configs/vBot_4.7/vBot/new_healer.otui [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+    margin-left: 9
+
+  Vocations
+    id: vocations
+    anchors.left: prev.right
+    margin-left: 5
+    anchors.verticalCenter: prev.verticalCenter
+
+Groups < FlatPanel
+  size: 150 90
```
Comment (kokekanon) [inline modules/game_bot/default_configs/vBot_4.7/vBot/new_healer.otui]:
```suggestion
  size: 150 100
```

Comment (kokekanon) [reply modules/game_bot/default_configs/vBot_4.7/vBot/new_healer.otui]: image


with 90 it is not possible to click on the checkbox

File: modules/game_bot/default_configs/vBot_4.7/vBot/new_healer.otui [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+    CategoryCheckBox
+      id: guild
+      text: Guild Members
+
+    CategoryCheckBox
+      id: botserver
+      text: BotServer Members
+  
+Vocations < FlatPanel
+  size: 100 90
```
Comment (kokekanon) [inline modules/game_bot/default_configs/vBot_4.7/vBot/new_healer.otui]:
```suggestion
  size: 100 100
```

File: modules/game_bot/default_configs/vBot_4.8/vBot/alarms.lua [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+  storage[panelName] = {}
+end
+
+local config = storage[panelName]
+
+ui.title:setOn(config.enabled)
+ui.title.onClick = function(widget)
+  config.enabled = not config.enabled
+  widget:setOn(config.enabled)
+end
```
Comment (kokekanon) [inline modules/game_bot/default_configs/vBot_4.8/vBot/alarms.lua]:
```suggestion
ui.title.onClick = function(widget)
  if not modules.client_options.getOption('enableAudio')  and not config.enabled then
   warning("Remember to activate sound in options ")
  end
  config.enabled = not config.enabled
  widget:setOn(config.enabled)
end
```

Comment (kokekanon) [reply modules/game_bot/default_configs/vBot_4.8/vBot/alarms.lua]: By clicking on "alarms" without having "enable audio" option activated. 
image

**Review by kokekanon — COMMENTED** [review]
Fixes that are within my range of knowledge

**EDIT**:  I made a mistake; I edited vbot 4.7 instead of vbot 4.8, sorry.
First time I'm doing this on GitHub xd

**Comment by mehah** [comment]
> Fixes that are within my range of knowledge
> 
> **EDIT**: I made a mistake; I edited vbot 4.7 instead of vbot 4.8, sorry. First time I'm doing this on GitHub xd

version 4.7 will be removed and wait for me to finish this PR so that the community can open new PR related to the bot.

**Comment by otmexa** [comment]
> > Fixes that are within my range of knowledge
> > **EDIT**: I made a mistake; I edited vbot 4.7 instead of vbot 4.8, sorry. First time I'm doing this on GitHub xd
> 
> version 4.7 will be removed and wait for me to finish this PR so that the community can open new PR related to the bot.

How is this implementation going? I'm dying to try it xD

**Comment by mehah** [comment]
> > > Fixes that are within my range of knowledge

finished at 85%, now it's up to the community, I'll do the merge soon.

**Comment by ericcobblepot** [comment]
> > > > Fixes that are within my range of knowledge
> 
> finished at 85%, now it's up to the community, I'll do the merge soon.

I think there is an error . the folders do not exist: 
- client_mods 
- game_bot

in the pr

**Comment by mehah** [comment]
> > > > > Fixes that are within my range of knowledge
> > 
> > 
> > finished at 85%, now it's up to the community, I'll do the merge soon.
> 
> I think there is an error . the folders do not exist:
> 
> * client_mods
> * game_bot
> 
> in the pr

sorry, mod has gitignore. fixed

**Pull request merged**